### ISSUE TITLE
enable debug for jemalloc in tests 

### ISSUE THREAD

**Issue by siddontang** [tikv/tikv#4227]
## Feature Request

**Is your feature request related to a problem? Please describe:**

Sometimes, we meet a core dump problem with a crashed stack, we have no way but only to guess there is something invalid memory usages like dangling pointer, or memory out of range. 

**Describe the feature you'd like:**

Maybe we can enable debug for jemalloc in tests, if we really have a memory problem, this may detect it. 

**Describe alternatives you've considered:**

One way is to use Valgrind, but I worry this may hurt the performance too much. 

Another way is to use sanitizer, see https://github.com/tikv/tikv/issues/1624

**Comment by aknuds1** [comment]
Do you need help with this?

**Comment by BusyJay** [comment]
Yes. @kennytm What do you think is better? sanitizer or jemalloc with debug enabled?

**Comment by kennytm** [comment]
Memory sanitizer (msan) doesn't work outside Linux and requires nightly, so I prefer something not involving msan.
